### PR TITLE
chore: add zvm post

### PR DESCRIPTION
**Pull request by hendriknielaender**

### CODE REVIEW

File: data/blog/zvm.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
+---
+publishDate: "September 22 2023"
+title: "Managing Zig Versions with zvm: A Technical Dive"
+description: "Explore the intricacies of building CLI applications in Zig, using zvm for version management, and the challenges of JSON parsing."
+image: ~/assets/images/thumbnails/zvm.png
```
Comment (flyck) [inline data/blog/zvm.md:5]: nit: chameleon is cropped weirdly when the post is opened. maybe a slight resize helps?

File: data/blog/zvm.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+While I won't be detailing the installation process or enumerating every feature of `zvm` in this post, I'll be unraveling some intriguing insights and takeaways from my journey of building it. From its core components to the nuanced challenges, let's delve into the world of **CLI applications in Zig** and version management.
+
+
+### A Simple Approach to CLI Command Parsing in Zig
+
+In building `zvm`, one of the key tasks was to implement a CLI (Command-Line Interface) that can effectively parse user input. This is crucial for a smooth user experience and for enabling the tool to act accordingly. Below is a simplified explanation of how I went about implementing CLI command parsing in Zig.
+
+#### Main Points:
+
+1\. **Data Structures**: I created custom structs, namely `CommandData` and `CommandOption`, to hold the relevant command and parameters.
```
Comment (flyck) [inline data/blog/zvm.md:31]: Nit: checked locally that these dont properly render after the recent theme changes. we need to add a bright color for these for `prose-strong`

File: data/blog/zvm.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+While Zig provides flexibility and performance, it does lean on the developer to handle many things manually. For JSON parsing, using or creating higher-level abstractions or libraries that facilitate JSON handling can help improve developer productivity and maintainability. Such utilities could offer:
+
+-   Helper functions to retrieve fields with type safety.
+-   More intuitive mechanisms for navigating nested JSON objects.
+-   Automated checks for missing or unexpected fields.
+
+
+## Wrap up
+---
```
Comment (flyck) [inline data/blog/zvm.md:193]: nit: looks bit weird but why not 👍🏻

File: data/blog/zvm.md (hunk 2 above)
Comment (flyck) [reply data/blog/zvm.md:31]: pushed the fix into this branch

**Comment by flyck** [comment]
Did the review, awesome stuff 💯

File: data/blog/zvm.md (hunk 1 above)
Comment (hendriknielaender) [reply data/blog/zvm.md]: 👍 did it, not perfect but works.

File: data/blog/zvm.md (hunk 3 above)
Comment (hendriknielaender) [reply data/blog/zvm.md]: removed ✅

**Pull request merged**